src/App.js

```diff
@@ -1,19 +1,38 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [messages, setMessages] = useState(chatMessages);
+  const [likes, setLikes] = useState(0);
+
+  const updateHeart = (messageToUpdate) => {
+    const updateMessages = messages.map(message => {
+      if (message.id === messageToUpdate.id) {
+        messageToUpdate.liked ? setLikes(likes - 1) : setLikes(likes + 1)
```

Comment:
Some more esoteric advice: In JavaScript, it's less expected to see a ternary used as a statement. An `if` statement is preferable when you have a statement and use a ternary for expressions. So if you want it to be a statement, that would look like:

```javascript
        if (messageToUpdate.liked) {
           setLikes(likes - 1)
         } else {
           setLikes(likes + 1)
         }
```

and if you want it to be an expression, you could do:

```javascript
        setLikes(likes + messageToUpdate.liked ? -1 : 1);
```

Reply:
In this case, though you might want to consider removing the `likes` state and calculate the number of likes when the component is rendered by iterating through `messages`.